MapApp/Controllers/CommentsController.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using MapApp.Data;
+using MapApp.Models.CommentsModels;
+using Microsoft.AspNetCore.Authorization;
+
+namespace MapApp.Controllers
+{
+    [Authorize]
+    public class CommentsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+
+        public CommentsController(ApplicationDbContext context)
+        {
+            _context = context;    
+        }
+
+        [Authorize(Roles = "Administrator")]
+        // GET: Comments
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Comment.ToListAsync());
+        }
+
+        // GET: Comments/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var comment = await _context.Comment
+                .SingleOrDefaultAsync(m => m.ID == id);
+            if (comment == null)
+            {
+                return NotFound();
+            }
+            return View(comment);
+        }
+
+        // GET: Comments/Create/2
+        public async Task<IActionResult> Create(int? loc_id)
+        {
+            if (loc_id == null)
+            {
+                return NotFound();
+            }
+
+            var comment = await _context.Location
+                .SingleOrDefaultAsync(m => m.ID == loc_id);
+            if (comment == null)
+            {
+                return NotFound();
+            }
+
+            ViewBag.loc_id = loc_id;
+            ViewBag.User = User.Identity.Name;
+            return View();
+        }
+
+        // POST: Comments/Create
+        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
+        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("ID,Header,Content,User,Location")] Comment comment)
+        {
+            if (ModelState.IsValid)
+            {
+                comment.CreateTime = DateTime.Now;
+                comment.EditTime = DateTime.Now;
```

Comment:
I think `CreateTime` and `EditTime` time should be exactly the same, e.g.:
```
comment.CreateTime = comment.EditTime = DateTime.Now;

Reply:
I changed them to be at the same line as you suggested.